ironic/drivers/modules/redfish/utils.py

```diff
@@ -274,3 +278,100 @@ def _get_system():
+                        retries=24):
+    if alive:
+        ping_type = "pingable"
+
+    else:
+        ping_type = "not pingable"
+
+    LOG.info("Waiting for the iDRAC to become %s", ping_type)
+
+    response_count = 0
+    state_reached = False
+
+    while retries > 0 and not state_reached:
+        response = _ping_host(host)
+        retries -= 1
+        if response == alive:
+            response_count += 1
+            LOG.debug("The iDRAC is %s, count=%s",
+                      ping_type,
+                      response_count)
+            if response_count == ping_count:
+                LOG.debug("Reached specified ping count")
+                state_reached = True
+        else:
+            response_count = 0
+            if alive:
+                LOG.debug("The iDRAC is still not pingable")
+            else:
+                LOG.debug("The iDRAC is still pingable")
+        time.sleep(10)
+
+    return state_reached
+
+def wait_for_host(redfish_node_ip):
+    LOG.debug("iDRAC was reset, waiting for return to operational state")
+    state_reached = wait_for_host_state(
+                            redfish_node_ip,
+                            alive=False,
+                            ping_count=2,
+                            retries=24)
+
+    if not state_reached:
+        raise exceptions.DRACOperationFailed(
+            drac_messages="Timed out waiting for the %s iDRAC to become "
+            "not pingable" % redfish_node_ip)
+
+    LOG.info("The iDRAC has become not pingable")
+
+    state_reached = wait_for_host_state(
+        redfish_node_ip,
+        alive=True,
+        ping_count=3,
+        retries=24)
+    if not state_reached:
+        raise exceptions.DRACOperationFailed(
+            drac_messages="Timed out waiting for the %s iDRAC to become "
+            "pingable" % redfish_node_ip)
+
+    LOG.info("The iDRAC has become pingable")
+
+def wait_until_idrac_is_ready(oem_manager, retries=None, retry_delay=None):
+    """Waits until the iDRAC is in a ready state
+    :param retries: The number of times to check if the iDRAC is
+                    ready. If None, the value of ready_retries that
+                    was provided when the object was created is
+                    used.
+    :param retry_delay: The number of seconds to wait between
+                        retries. If None, the value of
+                        ready_retry_delay that was provided when the
+                        object was created is used.
+    """
+
+    while retries > 0:
+        LOG.debug("Checking to see if the iDRAC is ready")
+
+        if oem_manager.is_idrac_ready():
+            LOG.debug("The iDRAC is ready")
+            return
+
+        LOG.debug("The iDRAC is not ready")
+        retries -= 1
+        if retries > 0:
+            time.sleep(retry_delay)
+
+    if retries == 0:
+        err_msg = "Timed out waiting for the iDRAC to become ready after reset"
+        LOG.error(err_msg)
+        raise
```

Comment:
We should raise a DracOperationError here with the err_msg

Reply:
Addressed, raised DracOperationError exception with err_msg